exercises/dnd-character/canonical-data.json

```diff
@@ -0,0 +1,170 @@
+{
+  "exercise": "dnd-character",
+  "version": "1.0.0",
+  "comments": [
+    "The random generator 'ability' can be property-tested.",
```

Comment:
It might be useful to link to an introduction of property-based testing, as that is a relatively unknown concept.

Reply:
Good idea. Fortunately there is https://hypothesis.works/articles/what-is-property-based-testing/ and Hypothesis is a cross-language property-based test framework, so if the (language-agnostic) article is not enough, it may even be the case for several language track that they want to use Hypothesis specifically.

For students on the Haskell track where property-based testing has come up as a subject, I've heard once or twice that people had prior experience with Hypothesis in, I think, Python.

Reply:
Would it make sense to link to this article in the description as well? Even though we expect to provide a sufficient test suite for each track, students as well as track implementers may be interested in knowing how such functions are tested, too.

Reply:
I think that including it in the description is a good idea. I also like the article.

Reply:
Hmm, now that I've tried, I don't think that I will. I think different languages have very different approaches to testing randomness. In Perl at work I sometimes use [`Test::MockRandom`](https://metacpan.org/pod/Test::MockRandom), and I can't find an equally language-agnostic article on mocking and dependency inversion.

So a cheap alternative is to let students inspect whatever tests implementers decide to make, and perhaps think about good linkable resources on how it's being tested and spread these later.